src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/JavaScriptEncoder.cs

```diff
@@ -205,4 +214,187 @@ private static unsafe bool TryWriteEncodedSingleCharacter(int unicodeScalar, cha
             return true;
         }
     }
+
+    internal sealed class UnsafeRelaxedJavaScriptEncoder : JavaScriptEncoder
+    {
+        private AllowedCharactersBitmap _allowedCharacters;
+
+        internal static readonly UnsafeRelaxedJavaScriptEncoder Singleton = new UnsafeRelaxedJavaScriptEncoder(new TextEncoderSettings(UnicodeRanges.All));
+
+        public UnsafeRelaxedJavaScriptEncoder(TextEncoderSettings filter)
+        {
+            if (filter == null)
+            {
+                throw new ArgumentNullException(nameof(filter));
+            }
+
+            _allowedCharacters = filter.GetAllowedCharacters();
+
+            // Forbid codepoints which aren't mapped to characters or which are otherwise always disallowed
+            // (includes categories Cc, Cs, Co, Cn, Zs [except U+0020 SPACE], Zl, Zp)
+            _allowedCharacters.ForbidUndefinedCharacters();
+
+            // '"' (U+0022 QUOTATION MARK) must always be escaped in Javascript / ECMAScript / JSON.
+            _allowedCharacters.ForbidCharacter('\"'); // can be used to escape attributes
+
+            // '\' (U+005C REVERSE SOLIDUS) must always be escaped in Javascript / ECMAScript / JSON.
+            // '/' (U+002F SOLIDUS) is not Javascript / ECMAScript / JSON-sensitive so doesn't need to be escaped.
+            _allowedCharacters.ForbidCharacter('\\');
+        }
+
+        public UnsafeRelaxedJavaScriptEncoder(params UnicodeRange[] allowedRanges) : this(new TextEncoderSettings(allowedRanges))
```

Comment:
The type is internal and used as a singleton, why are there two ctors?

Why are either of them declared `public`?

Reply:
Following the pattern used in `DefaultJavaScriptEncoder` which `UnsafeRelaxedJavaScriptEncoder` was copy-pasted from and modified. As a side effect, a lot of the quirks/formatting got retained. Do you think its worth updating (both)?

Reply:
I think that the new code should be held to the standards of new code, and ignore its origin.  I'd advise doing any cleanup on the old code as a followup change, particularly if this is expected to need to be cherry-picked into 3.0